src/ORM/Rule/ExistsIn.php

```diff
@@ -125,6 +126,51 @@ public function __invoke(EntityInterface $entity, array $options)
             }
         }
 
+        if (isset($options['parent']) && is_a($options['parent'], EntityInterface::class)) {
+            /**
+             * @var EntityInterface $parent
+             */
+            $parent = $options['parent'];
+
+            //extract entity from given parent regarding HM and HABTM associations
+            $parentEntity = null;
+            if ($isAssociation) {
+                if (TableRegistry::get($parent->getSource()) === $target->getTarget()) {
+                    //parent is of same  as target
+                    $parentEntity = $parent;
+                } else {
+                    if (!empty($parent[strtolower($target->getName())])) {
+                        $parentEntity = $parent[strtolower($target->getName())];
+                    }
+                }
+            } else {
+                $parentEntity = $parent;
+            }
+
+            if (isset($parentEntity)) {
+                //extract fields to match
+                $parent_fields = [];
+                if (is_array($parentEntity)) {
+                    //reagrd HM and HABTM associations
+                    $isNew = true;
+                    foreach ($parentEntity as $record) {
+                        $isNew &= $record->isNew();
```

Comment:
Do you need the reference here?

Reply:
This is not a reference it is the logic AND
```php
$isNew = $isNew & $record->isNew();
```

Reply:
But indeed it is wrong, it should be
```php
$isNew |= $record->isNew();
```